### PR TITLE
helm: Prevent duplicate enable-envoy-config

### LINKED ISSUE #23859: Helm Chart Renders Duplicate ConfigMap Keys When Ingress Controller, GatewayAPI, and/or L7LB Are Enabled
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

There are three Helm Values that will result in the `enable-envoy-config` key being added to Cilium's ConfigMap:

https://github.com/cilium/cilium/blob/66e5ec75ed8bbb96c4129634a75f01662cc439b3/install/kubernetes/cilium/templates/cilium-configmap.yaml#L217
https://github.com/cilium/cilium/blob/66e5ec75ed8bbb96c4129634a75f01662cc439b3/install/kubernetes/cilium/templates/cilium-configmap.yaml#L225
https://github.com/cilium/cilium/blob/66e5ec75ed8bbb96c4129634a75f01662cc439b3/install/kubernetes/cilium/templates/cilium-configmap.yaml#L206

If any two of these values are enabled simultaneously (such as Ingress Controller and GatewayAPI support), it results in a ConfigMap with duplicate keys. Helm itself does not care about duplicate keys, but other tools, such as Kustomize, do.

This could be rewritten something like this:

```sprig
{{- if or .Values.gatewayAPI.enabled .Values.ingressController.enabled .Values.loadBalancer.l7.backend }}
enable-envoy-config: "true"
{{- end }}
```

### Cilium Version

1.13.0

### Kernel Version

N/A

### Kubernetes Version

N/A

### Sysdump

N/A

### Relevant log output

```shell
N/A
```

### Anything else?

N/A

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by DWSR**
Please ensure your pull request adheres to the following guidelines:

- [x] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [x] All code is covered by unit and/or runtime tests where feasible.
- [x] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [x] If your commit description contains a `Fixes: <commit-id>` tag, then
      please add the commit author[s] as reviewer[s] to this issue.
- [x] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [x] Provide a title or release-note blurb suitable for the release notes.
- [x] Thanks for contributing!

Previously, if enabling any two of Ingress Controller, L7 Load Balancing, and GatewayAPI, the `cilium-config` ConfigMap would contain duplicate `enable-envoy-config` keys. Helm silently ignores this, but other Kubernetes tooling does not.
    
Fixes: #23859

```release-note
helm: Fix duplicate `enable-envoy-config` flag when enabling L7LB, Ingress Controller, or GatewayAPI simultaneously
```

### CODE REVIEW

**Review by sayboras — APPROVED** [review]
Thanks a lot for your issue and patch. LGTM :heavy_check_mark:

**Comment by DWSR** [comment]
@sayboras Looks like these test failures are unrelated to my change, but not sure. Any guidance?

**Comment by sayboras** [comment]
The failures are un-related to PR changes:

- test-1.16-4.19 hits https://github.com/cilium/cilium/issues/23845
- test-1.24-5.4) hits https://github.com/cilium/cilium/issues/22578
- test-runtime hits https://github.com/cilium/cilium/issues/22373

**Comment by DWSR** [comment]
@sayboras Anything else needed to merge this?

**Comment by sayboras** [comment]
> Anything else needed to merge this?

I don't think there is anything else required, except waiting folks to have another look and review. Once required codeowners approve, we can take it forward. Thanks for your contribution :+1:.

**Comment by sayboras** [comment]
Can you help to rebase with master? Otherwise, the merge commit will not have required sign-off (e.g. Signed-Off-By)

**Comment by DWSR** [comment]
> Can you help to rebase with master? Otherwise, the merge commit will not have required sign-off (e.g. Signed-Off-By)

Done. Let me know if you need anything else.

**Comment by sayboras** [comment]
Full CI is not required as the changes are checked as part of GHA jobs to make sure there is no regression.

**Comment by sayboras** [comment]
Integration test failure is due to the below error, which is re-related to the changes in this PR (mainly due to wrong sha checkout or merge commit).

```
diff --git a/Documentation/codeowners.rst b/Documentation/codeowners.rst
index b1ea129e..3857b5d7 100644
--- a/Documentation/codeowners.rst
+++ b/Documentation/codeowners.rst
@@ -168,6 +168,8 @@ external software and protocols:
 - `@cilium/sig-agent <https://github.com/orgs/cilium/teams/sig-agent>`__:
   Provide Cilium (agent) general Go review. Internal architecture, core data
   structures and daemon startup.
+- `@cilium/sig-bgp <https://github.com/orgs/cilium/teams/sig-bgp>`__:
+  Review changes to our BGP integration.
 - `@cilium/sig-clustermesh <https://github.com/orgs/cilium/teams/sig-clustermesh>`__:
   Ensure the reliability of state sharing between clusters to ensure that
   each cluster maintains a separate fault domain.
@@ -192,7 +194,7 @@ external software and protocols:
 - `@cilium/sig-lb <https://github.com/orgs/cilium/teams/sig-lb>`__:
   Maintain the layers necessary to coordinate all load balancing
   configurations within the agent control plane, including Services,
-  ClusterIP, NodePorts, Maglev, BGP export, local redirect policies, and
+  ClusterIP, NodePorts, Maglev, local redirect policies, and
   NAT46/NAT64.
 - `@cilium/sig-policy <https://github.com/orgs/cilium/teams/sig-policy>`__:
   Ensure consistency of semantics for all network policy representations.
HINT: to fix this, run 'make -C Documentation update-owners'
make[1]: *** [Makefile:74: check] Error 1
make: *** [Makefile:617: postcheck] Error 2
make: *** Waiting for unfinished jobs....
  CHECK   contrib/scripts/rand-check.sh
```

**Pull request merged**
